File: Visual Basic .NET/Theming/OrdersDetailDescription.vb
'
' * Xceed DataGrid for WPF - SAMPLE CODE - Theming Sample Application
' * Copyright (c) 2007-2024 Xceed Software Inc.
' * 
' * [OrdersDetailDescription.vb]
' * 
' * Custom detail description that returns the appropriate details for the
' * parent item received in the GetDetailsForParentItem override.
' *  
' * This file is part of the Xceed DataGrid for WPF product. The use 
' * and distribution of this Sample Code is subject to the terms 
' * and conditions referring to "Sample Code" that are specified in 
' * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
' *
' 

Imports System.Collections
Imports System.Data

Namespace Xceed.Wpf.DataGrid.Samples.Theming
  Public Class OrdersDetailDescription
	  Inherits DataGridDetailDescription
	Public Sub New()
	  Me.RelationName = "RelatedCustomers"
	End Sub

	Protected Overrides Function GetDetailsForParentItem(ByVal parentCollectionViewBase As DataGridCollectionViewBase, ByVal parentItem As Object) As IEnumerable
	  Dim parentRowView As DataRowView = CType(parentItem, DataRowView)

	  If parentRowView Is Nothing Then
		Return Nothing
	  End If

	  Dim CustomerID As String = CStr(parentRowView("CustomerID"))

	  Dim dataView As New DataView(parentRowView.Row.Table, "CustomerID = " & "'" & CustomerID & "'", Nothing, DataViewRowState.CurrentRows)

	  Return dataView
	End Function
  End Class
End Namespace

File: CSharp/Theming/OrdersDetailDescription.cs
/*
 * Xceed DataGrid for WPF - SAMPLE CODE - Theming Sample Application
 * Copyright (c) 2007-2024 Xceed Software Inc.
 * 
 * [OrdersDetailDescription.cs]
 * 
 * Custom detail description that returns the appropriate details for the
 * parent item received in the GetDetailsForParentItem override.
 *  
 * This file is part of the Xceed DataGrid for WPF product. The use 
 * and distribution of this Sample Code is subject to the terms 
 * and conditions referring to "Sample Code" that are specified in 
 * the XCEED SOFTWARE LICENSE AGREEMENT accompanying this product.
 *
 */

using System.Collections;
using System.Data;

namespace Xceed.Wpf.DataGrid.Samples.Theming
{
  public class OrdersDetailDescription : DataGridDetailDescription
  {
    public OrdersDetailDescription()
    {
      this.RelationName = "RelatedCustomers";
    }

    protected override IEnumerable GetDetailsForParentItem( DataGridCollectionViewBase parentCollectionViewBase, object parentItem )
    {
      DataRowView parentRowView = ( DataRowView )parentItem;

      if( parentRowView == null )
        return null;

      string CustomerID = ( string )parentRowView[ "CustomerID" ];

      DataView dataView = new DataView(
        parentRowView.Row.Table,
        "CustomerID = " + "'" + CustomerID + "'",
        null,
        DataViewRowState.CurrentRows );

      return dataView;
    }
  }
}
